spec/periodicChecksSpec.js

```diff
@@ -856,5 +856,61 @@ describe('Periodic Checks Module', () => {
         });
       });
     });
+
+    describe('Should Check If a Opened Issue is Assigned to a project', () => {
+      beforeEach(async () => {
+        github.issues.listForRepo = jasmine
+          .createSpy('listForRepo')
+          .and.resolveTo({
+            data: [
+              issues.withProject,
+              issues.anotherWithProject,
+              issues.withoutProject,
+            ],
+          });
+        payloadData.name = 'issues';
+        payloadData.payload.action = 'opened';
+        spyOn(periodicCheckModule, 'ensureNewIssuesHaveProjects').
+          and.callThrough();
+        await robot.receive(payloadData);
+      });
+
+      it('Should Call ensureNewIssuesHaveProjects function', () => {
+        expect(periodicCheckModule.ensureNewIssuesHaveProjects).
+          toHaveBeenCalled();
+      });
+      it('should get all project cards', () => {
+        expect(github.projects.listForRepo).toHaveBeenCalled();
+        expect(github.projects.listForRepo).toHaveBeenCalledWith({
+          repo: 'oppia',
+          owner: 'oppia',
+        });
```

Comment:
These tests look wrong.
Are these what we expect to happen?

Reply:
It looks right to me Because we have to first get all the project cards then filter out the ones required. Hence a test is required to check that all the project cards are fetched.
Looking forward to your reviews on this sir. Thanks.